### PR TITLE
Clean up code of legacy benchmark suite

### LINKED ISSUE #11076: Migrate benchmark pipeline to e2e test framework
To migrate the benchmark pipelines, we need to update the benchmark tools.

- [x] Upload e2e test artifacts in CI
- [x] Migrate Linux benchmark tool
- [x] Migrate compilation statistics tool
- [x] Migrate benchmark workflow to Github CI
- [x] Migrate Android benchmark tools
- [ ] Cleanup legacy code

### PR DESCRIPTION
**Pull request by pzread**
Clean up code of the legacy benchmark suite under `build_tools/benchmarks` and the related CMake rules. The legacy benchmark suite has been removed from repo 3 weeks ago.

I decided to put all cleanup changes in a single PR so it will be easier to revert, if needed. Each commit in this PR presents a single cleanup step and mostly focuses on a single area.

Fix #11076

### CODE REVIEW

File: build_tools/benchmarks/common/benchmark_driver.py [hunk 1]
```diff
@@ -87,76 +84,74 @@ def run(self) -> None:
...
-          self.benchmark_errors.append(e)
-          continue
-        finally:
-          # Some grace time.
-          time.sleep(self.benchmark_grace_time)
-
-        print("Benchmark completed")
-
-        if results_path:
+    benchmark_cases = self.benchmark_suite.filter_benchmarks(
```
Comment (pzread) [inline build_tools/benchmarks/common/benchmark_driver.py:87]: This part is the same as the deleted lines above, just changed the indent.

File: build_tools/benchmarks/common/benchmark_definition.py [hunk 2]
```diff
@@ -252,29 +233,18 @@ def __str__(self):
     params = ", ".join(params)
     return f"{self.platform_type.value} device <{params}>"
 
-  def get_iree_cpu_arch_name(self,
-                             use_legacy_name: bool = False) -> Optional[str]:
+  def get_iree_cpu_arch_name(
+      self) -> Optional[common_definitions.DeviceArchitecture]:
```
Comment (GMNGeoffrey) [inline build_tools/benchmarks/common/benchmark_definition.py]:
```suggestion
def get_iree_cpu_arch_name(self) -> Optional[common_definitions.DeviceArchitecture]:
```

(maybe just run autoformat)

File: build_tools/benchmarks/common/benchmark_definition.py [hunk 3]
```diff
@@ -252,29 +233,18 @@ def __str__(self):
...
 
-    arch = CPU_ABI_TO_TARGET_ARCH_MAP.get(name)
-    # TODO(#11076): Return common_definitions.DeviceArchitecture instead after
-    # removing the legacy path.
-    return None if arch is None else str(arch)
-
-  def get_iree_gpu_arch_name(self,
-                             use_legacy_name: bool = False) -> Optional[str]:
+  def get_iree_gpu_arch_name(
+      self) -> Optional[common_definitions.DeviceArchitecture]:
```
Comment (GMNGeoffrey) [inline build_tools/benchmarks/common/benchmark_definition.py]:
```suggestion
  def get_iree_gpu_arch_name(self) -> Optional[common_definitions.DeviceArchitecture]:
```

**Review by GMNGeoffrey — APPROVED** [review]

File: build_tools/benchmarks/common/benchmark_definition.py (hunk 2 above)
Comment (pzread) [reply build_tools/benchmarks/common/benchmark_definition.py]: `yapf` actually formatted them into two lines. I shortened the method names to make them fit into a single line.

Comment (GMNGeoffrey) [reply build_tools/benchmarks/common/benchmark_definition.py]: Oh weird. I guess my column counting is off

**Pull request merged**